File: Review/proyecto_reviews/lib/item.dart
// ignore_for_file: use_key_in_widget_constructors, prefer_const_constructors, unnecessary_new, sized_box_for_whitespace, avoid_unnecessary_containers, prefer_initializing_formals

import 'review.dart';

class Item {
  int idItem = 0;
  String tipo = "";
  String nombre = "";
  String genero = "";
  String autor = "";
  int duracion = 0;
  String rutaPortada = "";
  int estreno = 0;
  String sinopsis = "";
  double puntuacionMedia = 0.0;

  Item(
      int idItem,
      String tipo,
      String nombre,
      String genero,
      String autor,
      int duracion,
      String rutaPortada,
      int estreno,
      String sinopsis,
      double puntuacionMedia) {
    this.idItem = idItem;
    this.tipo = tipo;
    this.nombre = nombre;
    this.genero = genero;
    this.autor = autor;
    this.duracion = duracion;
    this.rutaPortada = rutaPortada;
    this.estreno = estreno;
    this.sinopsis = sinopsis;
    this.puntuacionMedia = puntuacionMedia;
  }

  void setIdItem(int idItem) {
    this.idItem = idItem;
  }

  int getIdItem() {
    return idItem;
  }

  void setTipo(String tipo) {
    this.tipo = tipo;
  }

  String getTipo() {
    return tipo;
  }

  void setNombre(String nombre) {
    this.nombre = nombre;
  }

  String getNombre() {
    return nombre;
  }

  void setGenero(String genero) {
    this.genero = genero;
  }

  String getGenero() {
    return genero;
  }

  void setAutor(String autor) {
    this.autor = autor;
  }

  String getAutor() {
    return autor;
  }

  void setDuracion(int duracion) {
    this.duracion = duracion;
  }

  int getDuracion() {
    return duracion;
  }

  void setRutaPortada(String rutaPortada) {
    this.rutaPortada = rutaPortada;
  }

  String getRutaPortada() {
    return rutaPortada;
  }

  void setEstreno(int estreno) {
    this.estreno = estreno;
  }

  int getEstreno() {
    return estreno;
  }

  void setSinopsis(String sinopsis) {
    this.sinopsis = sinopsis;
  }

  String getSinopsis() {
    return sinopsis;
  }

  void setPuntuacionMedia(double puntuacionMedia) {
    this.puntuacionMedia = puntuacionMedia;
  }

  double getPuntuacionMedia() {
    return puntuacionMedia;
  }
}

File: Review/apiPruebas/src/main/java/com/example/ReviewsProyecto/model/Item.java
package com.example.ReviewsProyecto.model;

import jakarta.persistence.*;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import java.io.Serializable;
import java.util.Set;



@Entity
@Table(name = "Items")
public class Item implements Serializable {

    private int idItem;
    private String tipo;
    private String nombre;
    private String genero;
    private String autor;
    private int duracion;
    private String rutaPortada;
    private int estreno;
    private String sinopsis;
    private Set<Review> reviewSet;

    public Item() {
    }

    public Item(int idItem, String tipo, String nombre, String genero, String autor, int duracion, String rutaPortada, int estreno, String sinopsis, Set<Review> reviewSet) {
        this.idItem = idItem;
        this.tipo = tipo;
        this.nombre = nombre;
        this.genero = genero;
        this.autor = autor;
        this.duracion = duracion;
        this.rutaPortada = rutaPortada;
        this.estreno = estreno;
        this.sinopsis = sinopsis;
        this.reviewSet = reviewSet;
    }


    @Id
    @Column(name = "ItemID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int getIdItem() {
        return idItem;
    }


    public void setIdItem(int idItem) {
        this.idItem = idItem;
    }

    @Column(name = "Tipo")
    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }
    @Column(name = "Nombre")
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Column(name = "Genero")
    public String getGenero() {
        return genero;
    }

    public void setGenero(String genero) {
        this.genero = genero;
    }

    @Column(name = "Autor")
    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    @Column(name = "Duracion")
    public int getDuracion() {
        return duracion;
    }

    public void setDuracion(int duracion) {
        this.duracion = duracion;
    }

    @Column(name = "RutaPortada")
    public String getRutaPortada() {
        return rutaPortada;
    }

    public void setRutaPortada(String rutaPortada) {
        this.rutaPortada = rutaPortada;
    }

    @Column(name = "Estreno")
    public int getEstreno() {
        return estreno;
    }

    public void setEstreno(int estreno) {
        this.estreno = estreno;
    }

    @Column(name = "Sinopsis")
    public String getSinopsis() {
        return sinopsis;
    }

    public void setSinopsis(String sinopsis) {
        this.sinopsis = sinopsis;
    }

    @OneToMany(mappedBy = "item")
    @Fetch(FetchMode.JOIN)
    public Set<Review> getReviewSet() {
        return reviewSet;
    }

    public void setReviewSet(Set<Review> reviewSet) {
        this.reviewSet = reviewSet;
    }

    @Override
    public String toString() {
        return "Item{" +
                "idItem=" + idItem +
                ", tipo='" + tipo + '\'' +
                ", nombre='" + nombre + '\'' +
                ", genero='" + genero + '\'' +
                ", autor='" + autor + '\'' +
                ", duracion=" + duracion +
                ", rutaPortada='" + rutaPortada + '\'' +
                ", estreno=" + estreno +
                ", sinopsis='" + sinopsis + '\'' +
                '}';
    }
}
